statistics/histogram.go

```diff
@@ -834,29 +875,73 @@ func (c *Column) IsInvalid(sc *stmtctx.StatementContext, collPseudo bool) bool {
 	if collPseudo && c.NotAccurate() {
 		return true
 	}
-	if c.NDV > 0 && c.Len() == 0 && sc != nil {
+	// If stats ver < 2, histogram represents all data in this column, so empty histogram implies unloaded stats.
+	// In stats ver 2, histogram + TopN represents all data in a column, so both of them being empty implies unloaded stats.
+	if c.NDV > 0 && sc != nil &&
+		((c.Len() == 0 && c.StatsVer < Version2) || (c.Len() == 0 && c.TopN.TotalCount() == 0 && c.StatsVer == Version2)) {
 		sc.SetHistogramsNotLoad()
 		HistogramNeededColumns.insert(tableColumnID{TableID: c.PhysicalID, ColumnID: c.Info.ID})
 	}
-	return c.TotalRowCount() == 0 || (c.NDV > 0 && c.Len() == 0)
+	return c.TotalRowCount() == 0 ||
+		(c.NDV > 0 &&
+			((c.Len() == 0 && c.StatsVer < Version2) || (c.Len() == 0 && c.TopN.TotalCount() == 0 && c.StatsVer == Version2)))
```

Comment:
We can impl a `notNullCount` for column and then directly use it here to simplify the checking?

Reply:
done.